<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Proportional Venn and UpSet Diagrams for Gene Sets and Genomic Regions • gVenn</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Proportional Venn and UpSet Diagrams for Gene Sets and Genomic Regions">
<meta name="description" content="Tools to compute and visualize overlaps between gene sets or genomic regions. Venn diagrams with proportional areas are provided, while UpSet plots are recommended for larger numbers of sets. The package supports GRanges and GRangesList inputs, and integrates with analysis workflows for ChIP-seq, ATAC-seq, and other genomic interval data. It generates clean, interpretable, and publication-ready figures.">
<meta property="og:description" content="Tools to compute and visualize overlaps between gene sets or genomic regions. Venn diagrams with proportional areas are provided, while UpSet plots are recommended for larger numbers of sets. The package supports GRanges and GRangesList inputs, and integrates with analysis workflows for ChIP-seq, ATAC-seq, and other genomic interval data. It generates clean, interpretable, and publication-ready figures.">
<!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZHJNLPW1JE"></script><script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-ZHJNLPW1JE'); </script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">gVenn</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/gVenn.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ckntav/gVenn/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gvenn">gVenn<a class="anchor" aria-label="anchor" href="#gvenn"></a>
</h1></div>
<p><img src="https://img.shields.io/badge/maintainer-ckntav-blue" alt="Maintainer"><img src="https://img.shields.io/github/last-commit/ckntav/gVenn" alt="Last commit"><a href="https://doi.org/10.5281/zenodo.17918547" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.17918547.svg" alt="DOI"></a></p>
<p><strong>Proportional Venn diagrams for genomic regions and gene set overlaps</strong></p>
<p align="center">
<img src="reference/figures/gVenn_hex_sticker.png" width="175"></p>
<p><strong>gVenn</strong> stands for <strong>gene/genomic Venn</strong>.<br>
It provides tools to compute overlaps between genomic regions or sets of genes and visualize them as Venn diagrams with areas proportional to the number of overlapping elements. With seamless support for <code>GRanges</code> and <code>GRangesList</code> objects, <strong>gVenn</strong> integrates naturally into Bioconductor workflows such as ChIP-seq, ATAC-seq, or other interval-based analyses, and produces clean, publication-ready figures.</p>
<p align="center">
<img src="reference/figures/Tav_graphical_abstract_v3_20251029.png" width="800"></p>
<blockquote>
<p>The gVenn hex logo was generated using the <a href="https://github.com/GuangchuangYu/hexSticker" class="external-link">hexSticker</a> R package (<a href="https://github.com/ckntav/gVenn_manuscript/blob/main/scripts/generate_hex_sticker_logo/02_generate_hex_sticker.R" class="external-link">code</a>)</p>
</blockquote>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The gVenn package is available through Bioconductor and GitHub.</p>
<p>You can install it from Bioconductor using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"gVenn"</span><span class="op">)</span></span></code></pre></div>
<p>To install the development version from GitHub, use:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")  # if not already installed</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"ckntav/gVenn"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># or, alternatively:</span></span>
<span><span class="co"># install.packages("devtools")  # if not already installed</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ckntav/gVenn"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>For comprehensive documentation and vignettes, visit the <a href="https://ckntav.github.io/gVenn/">gVenn documentation page</a>.</p>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>This quick example demonstrates how to compute overlaps between ChIP-seq peaks and visualize them with both a Venn diagram and an UpSet plot.</p>
<div class="section level3">
<h3 id="id_1-load-example-chip-seq-data-and-compute-overlaps">1. Load example ChIP-seq data and compute overlaps<a class="anchor" aria-label="anchor" href="#id_1-load-example-chip-seq-data-and-compute-overlaps"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ckntav/gVenn" class="external-link">gVenn</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Example dataset of ChIP-seq peaks (A549 cell line, 3 set of genomic regions)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute overlaps</span></span>
<span><span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/computeOverlaps.html">computeOverlaps</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required namespace: GenomicRanges</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_2-visualize">2. Visualize<a class="anchor" aria-label="anchor" href="#id_2-visualize"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Draw Venn diagram</span></span>
<span><span class="fu"><a href="reference/plotVenn.html">plotVenn</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-example_venn-1.png" alt="" width="100%"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Draw UpSet plot (useful for larger numbers of sets)</span></span>
<span><span class="fu"><a href="reference/plotUpSet.html">plotUpSet</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-example_upset-1.png" alt="" width="100%"></p>
</div>
<div class="section level3">
<h3 id="id_3-extract-elements-per-overlap-group">3. Extract elements per overlap group<a class="anchor" aria-label="anchor" href="#id_3-extract-elements-per-overlap-group"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extractOverlaps.html">extractOverlaps</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display the number of genomic regions per overlap group</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">groups</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt; group_010 group_001 group_100 group_110 group_011 group_101 group_111 </span></span>
<span><span class="co">#&gt;       267       125         4        48        46        16       243</span></span></code></pre></div>
<p>In this example:</p>
<ul>
<li>243 peaks are shared across all three factors (MED1, BRD4, and GR)</li>
<li>267 peaks are unique to BRD4</li>
<li>48 peaks are shared between MED1 and BRD4 only</li>
</ul>
<div class="section level4">
<h4 id="overlap-group-naming">Overlap group naming<a class="anchor" aria-label="anchor" href="#overlap-group-naming"></a>
</h4>
<p>When overlaps are computed, each group of elements or genomic regions is labeled with a binary code that indicates which sets the element belongs to.</p>
<ul>
<li>Each digit in the code corresponds to one input set (e.g., A, B, C).</li>
<li>A 1 means the element is present in that set, while 0 means absent.</li>
<li>The group names in the output are prefixed with “group_” for clarity.</li>
</ul>
<div align="center">
<table class="table">
<thead><tr class="header">
<th>Group name</th>
<th>Meaning</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>group_100</code></td>
<td>Elements only in <strong>A</strong>
</td>
</tr>
<tr class="even">
<td><code>group_010</code></td>
<td>Elements only in <strong>B</strong>
</td>
</tr>
<tr class="odd">
<td><code>group_001</code></td>
<td>Elements only in <strong>C</strong>
</td>
</tr>
<tr class="even">
<td><code>group_110</code></td>
<td>Elements in <strong>A ∩ B</strong> (not C)</td>
</tr>
<tr class="odd">
<td><code>group_101</code></td>
<td>Elements in <strong>A ∩ C</strong> (not B)</td>
</tr>
<tr class="even">
<td><code>group_011</code></td>
<td>Elements in <strong>B ∩ C</strong> (not A)</td>
</tr>
<tr class="odd">
<td><code>group_111</code></td>
<td>Elements in <strong>A ∩ B ∩ C</strong>
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level4">
<h4 id="retrieve-one-particular-overlap-group">Retrieve one particular overlap group<a class="anchor" aria-label="anchor" href="#retrieve-one-particular-overlap-group"></a>
</h4>
<p>Each overlap group can be accessed directly by name for downstream analyses, including motif enrichment, transcription factor (TF) enrichment, annotation of peaks to nearby genes, functional enrichment or visualization.</p>
<p>For example, to extract all elements that are present in <strong>A ∩ B ∩ C</strong>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract elements in group_111 (present in all three sets: MED1_Dex_chr7, BRD4_Dex_chr7, GR_Dex_chr7)</span></span>
<span><span class="va">peaks_in_all_sets</span> <span class="op">&lt;-</span> <span class="va">groups</span><span class="op">[[</span><span class="st">"group_111"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Display the elements</span></span>
<span><span class="va">peaks_in_all_sets</span></span>
<span><span class="co">#&gt; GRanges object with 243 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;         seqnames              ranges strand | intersect_category</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]     chr7     1156721-1157555      * |                111</span></span>
<span><span class="co">#&gt;     [2]     chr7     1520256-1521263      * |                111</span></span>
<span><span class="co">#&gt;     [3]     chr7     2309811-2310529      * |                111</span></span>
<span><span class="co">#&gt;     [4]     chr7     3027924-3028466      * |                111</span></span>
<span><span class="co">#&gt;     [5]     chr7     3436651-3437214      * |                111</span></span>
<span><span class="co">#&gt;     ...      ...                 ...    ... .                ...</span></span>
<span><span class="co">#&gt;   [239]     chr7 158431413-158433728      * |                111</span></span>
<span><span class="co">#&gt;   [240]     chr7 158818200-158819318      * |                111</span></span>
<span><span class="co">#&gt;   [241]     chr7 158821076-158821876      * |                111</span></span>
<span><span class="co">#&gt;   [242]     chr7 158863108-158864616      * |                111</span></span>
<span><span class="co">#&gt;   [243]     chr7 159015311-159016245      * |                111</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>Pull requests are welcome. If you find a bug, have a suggestion, or want to add a feature, please open an issue first to discuss what you would like to change.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/gVenn" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/ckntav/gVenn/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ckntav/gVenn/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gVenn</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://ckntav.github.io/" class="external-link">Christophe Tav</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8808-9617" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://ckntav.github.io/" class="external-link">Christophe Tav</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
