<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Extract Overlap Groups from Genomic or Set Overlap Results — extractOverlaps • gVenn</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Extract Overlap Groups from Genomic or Set Overlap Results — extractOverlaps"><meta name="description" content='This function extracts subsets of intersecting elements grouped by their
overlap category (e.g., "110").
For genomic overlaps, it returns a GRangesList; for set overlaps, it
returns a named list of character vectors.'><meta property="og:description" content='This function extracts subsets of intersecting elements grouped by their
overlap category (e.g., "110").
For genomic overlaps, it returns a GRangesList; for set overlaps, it
returns a named list of character vectors.'><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZHJNLPW1JE"></script><script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-ZHJNLPW1JE'); </script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gVenn</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gVenn.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ckntav/gVenn/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Extract Overlap Groups from Genomic or Set Overlap Results</h1>
      <small class="dont-index">Source: <a href="https://github.com/ckntav/gVenn/blob/master/R/extractOverlaps.R" class="external-link"><code>R/extractOverlaps.R</code></a></small>
      <div class="d-none name"><code>extractOverlaps.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function extracts subsets of intersecting elements grouped by their
overlap category (e.g., "110").
For genomic overlaps, it returns a <code>GRangesList</code>; for set overlaps, it
returns a named list of character vectors.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">extractOverlaps</span><span class="op">(</span><span class="va">overlap_object</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-overlap-object">overlap_object<a class="anchor" aria-label="anchor" href="#arg-overlap-object"></a></dt>
<dd><p>A <code>GenomicOverlapsResult</code> or <code>SetOverlapsResult</code>
object.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A named list of grouped intersecting elements:</p><ul><li><p>If input is a <code>GenomicOverlapsResult</code>, a <code>GRangesList</code> split
by <code>intersect_category</code>.</p></li>
<li><p>If input is a <code>SetOverlapsResult</code>, a named <code>list</code> of character vectors
grouped by <code>intersect_category</code>.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example with gene sets (built-in dataset)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="computeOverlaps.html">computeOverlaps</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">group_gene</span> <span class="op">&lt;-</span> <span class="fu">extractOverlaps</span><span class="op">(</span><span class="va">res_sets</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">group_gene</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ACTN1"  "ALDOA"  "CRYBG1" "AK4"    "ACYP2"  "AFP"    "ACTN2"  "AMPD3" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "ACADS"  "ACY1"   "ADH1B"  "ACTBP7" "ADH6"   "ACTBP4" "ANXA2"  "AGXT"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] "TLE5"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "AFM"       "ADPRH"     "AIF1"      "ACVR2A"    "ACTA1"     "PLIN2"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ALDH1A3"   "ALK"       "ACTG2"     "ADCY8"     "ABR"       "ADCYAP1"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ADRA2A"    "ADORA2B"   "ADRA1B"    "ANXA5"     "ACLY"      "ANK3"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ADCY6"     "ACR"       "ADAM10"    "AARS1"     "ACVR2B"    "ACO2"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ADH1C"     "PARP1"     "ADD2"      "ADARB1"    "ADCY2"     "ANXA1"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "ALCAM"     "ADORA2A"   "AMFR"      "AMBN"      "NAT2"      "A1BG"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [37] "ACADL"     "ADH5"      "ACTG1P8"   "AADAC"     "ACOX1"     "ALDH9A1"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [43] "ANXA2P2"   "ADORA2BP1" "ADRB3"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ALPP"     "ACTG1P9"  "AHSG"     "ASIC2"    "ACTG1P10" "ALAS1"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "AKT2"     "PARP1P1"  "ABCD1"    "SLC25A6"  "AAMP"     "ADCP1"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ACADVL"   "ACTG1"    "ANGPT2"   "AGTR1"    "ACACB"    "ACTBP9"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ALDH1B1"  "ADAR"     "ABCD2"    "AMHR2"    "ABCB7"    "ABCA1"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "PARP4"    "ACTG1P1"  "JAG1"     "ACTA2"    "ADH7"     "AP1B1"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "ACVR1"    "ACTN4"    "A2MP1"    "ABCA4"    "ALAD"     "ADRA1A"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [37] "ADCY5"    "ALDOB"    "AP2B1"    "AMELY"    "ABL1"     "ACTC1"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [43] "AK2"      "ALOX12B"  "ACTN3"    "AIC"      "ALB"      "NATP"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [49] "ANG"      "AHR"      "ABCA2"    "ALPL"     "ANXA2P1"  "AMELX"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [55] "AHCY"     "PARP1P2"  "ALOX5"    "AMPD1"    "AFA"      "ACADSB"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [61] "AIH3"     "ACAN"     "AGA"      "AMY1C"    "ADSS2"    "ALDH2"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [67] "ALOX15B" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "AMY1A"  "ACRV1"  "ALAS2"  "ABCA3"  "ALPG"   "AMY2A"  "ADH4"   "ADARB2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "NR0B1"  "AMYP1"  "A2M"    "AGTR2"  "ALOX15" "ACP1"   "ADH1A"  "AF8T"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "NAT1"   "ACAA1"  "AGT"    "AMD1P2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ADRB1"    "ABAT"     "ALOX5AP"  "ADD1"     "ACVR1B"   "AANAT"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ADSL"     "ADCY7"    "ALX3"     "ALOX12P1" "ANGPT1"   "ACTG1P6" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ADAM8"    "ACHE"     "ANCR"     "ACP3"     "ACP5"     "APLNR"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ACTBP8"   "ADCY1"    "ADA"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ACP2"      "ALDH3A1"   "ACTB"      "ACACA"     "ASIC1"     "SLC25A5"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ACTL6A"    "AMY2B"     "AMH"       "AMPH"      "ADK"       "ALDH3A2"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ACTG1P3"   "ACO1"      "ACTG1P7"   "ALPI"      "ANXA4"     "AGL"      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ADRB2"     "ABCF1"     "ABO"       "AMD1"      "ALS3"      "ALOX12"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "AMBP"      "AMPD2"     "ALDH1A1"   "AFG3L1P"   "ADFN"      "ADCYAP1R1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "ADD3"      "ALOX12P2"  "BIN1"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example with genomic regions (built-in dataset)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res_genomic</span> <span class="op">&lt;-</span> <span class="fu"><a href="computeOverlaps.html">computeOverlaps</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">group_genomic</span> <span class="op">&lt;-</span> <span class="fu">extractOverlaps</span><span class="op">(</span><span class="va">res_genomic</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">group_genomic</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRangesList object of length 7:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $group_010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 267 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames              ranges strand | intersect_category</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr7       234690-235402      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr7       538240-538633      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr7     1504294-1504733      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr7     1506830-1507301      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr7     1513353-1513690      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...                 ...    ... .                ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [263]     chr7 155618941-155619523      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [264]     chr7 155644241-155644737      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [265]     chr7 158829343-158830028      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [266]     chr7 158856251-158856723      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [267]     chr7 159012435-159013222      * |                010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;6 more elements&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://ckntav.github.io/" class="external-link">Christophe Tav</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

