<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>gVenn: Proportional Venn diagrams for genomic regions and gene set overlaps • gVenn</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="gVenn: Proportional Venn diagrams for genomic regions and gene set overlaps">
<!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZHJNLPW1JE"></script><script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-ZHJNLPW1JE'); </script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gVenn</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/gVenn.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ckntav/gVenn/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>gVenn: Proportional Venn diagrams for genomic regions and gene set overlaps</h1>
                        <h4 data-toc-skip class="author">Christophe
Tav</h4>
            
            <h4 data-toc-skip class="date">August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ckntav/gVenn/blob/master/vignettes/gVenn.Rmd" class="external-link"><code>vignettes/gVenn.Rmd</code></a></small>
      <div class="d-none name"><code>gVenn.Rmd</code></div>
    </div>

    
    
<p><img src="figures/20250827_hex_gVenn_v1.png" width="20%" style="display: block; margin: auto;"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>gVenn</strong> stands for <strong>gene/genomic
Venn</strong>.<br>
It provides tools to compute overlaps between genomic regions or sets of
genes and visualize them as <strong>Venn</strong> diagrams with areas
proportional to the number of overlapping elements. In addition, the
package can generate <strong>UpSet</strong> plots for cases with many
sets, offering a clear alternative to complex Venn diagrams.</p>
<p>With seamless support for <code>GRanges</code> and
<code>GRangesList</code> objects, <strong>gVenn</strong> integrates
naturally into Bioconductor workflows such as ChIP-seq, ATAC-seq, or
other interval-based analyses.</p>
<p>Overlap groups can be easily extracted for further analysis, such as
motif enrichment, transcription factor binding enrichment, or gene
annotation. <strong>gVenn</strong> package produces clean,
publication-ready figures.</p>
<p><br><img src="figures/20250827_graphical_abstract_v2.png" width="100%" style="display: block; margin: auto;"><br></p>
</div>
<div class="section level2">
<h2 id="example-workflow">Example workflow<a class="anchor" aria-label="anchor" href="#example-workflow"></a>
</h2>
<p>This section demonstrates a typical workflow with gVenn, from
computing overlaps to generating clean, publication-ready figures. The
examples show how to work with genomic interval data.</p>
<p>We start by loading the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ckntav/gVenn" class="external-link">gVenn</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-example-chip-seq-genomic-regions">1. Load example ChIP-seq genomic regions<a class="anchor" aria-label="anchor" href="#load-example-chip-seq-genomic-regions"></a>
</h3>
<p>We use the dataset
<strong><code>a549_chipseq_dataset</code></strong>, which contains
example consensus peak subsets for <strong>MED1</strong>,
<strong>BRD4</strong>, and <strong>GR</strong> after dexamethasone
treatment in A549 cells. To keep the dataset small and suitable for
examples and tests, each set has been restricted to peaks located on
<em>chromosome 7</em>.</p>
<p>These data originate from Tav <em>et al.</em> (2023)
<a href="https://doi.org/10.3389/fgene.2023.1237092" target="_blank" class="external-link">
doi:10.3389/fgene.2023.1237092</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example A549 ChIP-seq peaks (subset on chr7 for demo)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compute-overlaps-between-genomic-regions">2. Compute overlaps between genomic regions<a class="anchor" aria-label="anchor" href="#compute-overlaps-between-genomic-regions"></a>
</h3>
<p>We compute overlaps between the ChIP-seq peak sets using
<code><a href="../reference/computeOverlaps.html">computeOverlaps()</a></code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genomic_overlaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeOverlaps.html">computeOverlaps</a></span><span class="op">(</span><span class="va">a549_chipseq_peaks</span><span class="op">)</span></span></code></pre></div>
<p>The result is a structured object that contains</p>
<ul>
<li>A GRanges object, where each region includes metadata describing its
overlap pattern across the input sets.</li>
<li>An associated count matrix (or data frame) summarizing the number of
regions in each intersection.</li>
</ul>
</div>
<div class="section level3">
<h3 id="visualization">3. Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<div class="section level4">
<h4 id="venn-diagram">Venn diagram<a class="anchor" aria-label="anchor" href="#venn-diagram"></a>
</h4>
<p><code><a href="../reference/plotVenn.html">plotVenn()</a></code> draws proportional Venn diagrams from the
overlap object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotVenn.html">plotVenn</a></span><span class="op">(</span><span class="va">genomic_overlaps</span><span class="op">)</span></span></code></pre></div>
<p><img src="gVenn_files/figure-html/plot_venn-1.png" width="480" style="display: block; margin: auto;"><br></p>
</div>
<div class="section level4">
<h4 id="upset-plot">UpSet plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h4>
<p>For more than <strong>three sets</strong>, a Venn diagram with
<strong>areas exactly proportional</strong> to all intersections is
<strong>generally not mathematically attainable</strong>. Solvers (like
those used by <code>eulerr</code>) provide <strong>best-effort
approximations</strong>, but the layout can become hard to read. In
these cases, an <strong>UpSet plot</strong> is the recommended
visualization because it scales cleanly to many sets and preserves
intersection sizes precisely on bar axes.</p>
<p>We therefore suggest using <code><a href="../reference/plotUpSet.html">plotUpSet()</a></code> when you have
<strong>&gt; 3 sets</strong> (or any time the Venn becomes visually
crowded).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotUpSet.html">plotUpSet</a></span><span class="op">(</span><span class="va">genomic_overlaps</span><span class="op">)</span></span></code></pre></div>
<p><img src="gVenn_files/figure-html/plot_upset-1.png" width="480" style="display: block; margin: auto;"><br></p>
</div>
<div class="section level4">
<h4 id="export-visualization">Export visualization<a class="anchor" aria-label="anchor" href="#export-visualization"></a>
</h4>
<p>You can export any visualization using <code><a href="../reference/saveViz.html">saveViz()</a></code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">venn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotVenn.html">plotVenn</a></span><span class="op">(</span><span class="va">genomic_overlaps</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/saveViz.html">saveViz</a></span><span class="op">(</span><span class="va">venn</span>,</span>
<span>        output_dir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>        output_file <span class="op">=</span> <span class="st">"figure_gVenn"</span>,</span>
<span>        format <span class="op">=</span> <span class="st">"pdf"</span><span class="op">)</span></span></code></pre></div>
<p>By default, files are written to the current directory (“.”). If you
enabled the date option (today), the current date will be prepended to
the filename.</p>
<p>You can also export to PNG or SVG:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/saveViz.html">saveViz</a></span><span class="op">(</span><span class="va">venn</span>,</span>
<span>        output_dir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>        output_file <span class="op">=</span> <span class="st">"figure_gVenn"</span>,</span>
<span>        format <span class="op">=</span> <span class="st">"png"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/saveViz.html">saveViz</a></span><span class="op">(</span><span class="va">venn</span>,</span>
<span>        output_dir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>        output_file <span class="op">=</span> <span class="st">"figure_gVenn"</span>,</span>
<span>        format <span class="op">=</span> <span class="st">"svg"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="extract-elements-per-overlap-group">4. Extract elements per overlap group<a class="anchor" aria-label="anchor" href="#extract-elements-per-overlap-group"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extractOverlaps.html">extractOverlaps</a></span><span class="op">(</span><span class="va">genomic_overlaps</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display the number of genomic regions per overlap group</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">groups</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="co">#&gt; group_010 group_001 group_100 group_110 group_011 group_101 group_111 </span></span>
<span><span class="co">#&gt;       267       125         4        48        46        16       243</span></span></code></pre></div>
<p><br></p>
<div class="section level5">
<h5 id="overlap-group-naming">Overlap group naming<a class="anchor" aria-label="anchor" href="#overlap-group-naming"></a>
</h5>
<p>When overlaps are computed, each group of elements or genomic regions
is labeled with a binary code that indicates which sets the element
belongs to.</p>
<ul>
<li>Each digit in the code corresponds to one input set (e.g., A, B,
C).</li>
<li>A 1 means the element is present in that set, while 0 means
absent.</li>
<li>The group names in the output are prefixed with “group_” for
clarity.</li>
</ul>
<div align="center" width="50%">
<table class="table">
<thead><tr class="header">
<th>Group name</th>
<th>Meaning</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>group_100</code></td>
<td>Elements only in <strong>A</strong>
</td>
</tr>
<tr class="even">
<td><code>group_010</code></td>
<td>Elements only in <strong>B</strong>
</td>
</tr>
<tr class="odd">
<td><code>group_001</code></td>
<td>Elements only in <strong>C</strong>
</td>
</tr>
<tr class="even">
<td><code>group_110</code></td>
<td>Elements in <strong>A ∩ B</strong> (not C)</td>
</tr>
<tr class="odd">
<td><code>group_101</code></td>
<td>Elements in <strong>A ∩ C</strong> (not B)</td>
</tr>
<tr class="even">
<td><code>group_011</code></td>
<td>Elements in <strong>B ∩ C</strong> (not A)</td>
</tr>
<tr class="odd">
<td><code>group_111</code></td>
<td>Elements in <strong>A ∩ B ∩ C</strong>
</td>
</tr>
</tbody>
</table>
</div>
<p><br></p>
</div>
<div class="section level5">
<h5 id="extract-one-particular-group">Extract one particular group<a class="anchor" aria-label="anchor" href="#extract-one-particular-group"></a>
</h5>
<p>Each overlap group can be accessed directly by name for downstream
analyses, including motif enrichment, transcription factor (TF)
enrichment, annotation of peaks to nearby genes, functional enrichment
or visualization.</p>
<p>For example, to extract all elements that are present in <strong>A ∩
B ∩ C</strong>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract elements in group_111 (present in A, B, and C)</span></span>
<span><span class="va">peaks_in_all_sets</span> <span class="op">&lt;-</span> <span class="va">groups</span><span class="op">[[</span><span class="st">"group_111"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Display the elements</span></span>
<span><span class="va">peaks_in_all_sets</span></span>
<span><span class="co">#&gt; GRanges object with 243 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;         seqnames              ranges strand | intersect_category</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]     chr7     1156721-1157555      * |                111</span></span>
<span><span class="co">#&gt;     [2]     chr7     1520256-1521263      * |                111</span></span>
<span><span class="co">#&gt;     [3]     chr7     2309811-2310529      * |                111</span></span>
<span><span class="co">#&gt;     [4]     chr7     3027924-3028466      * |                111</span></span>
<span><span class="co">#&gt;     [5]     chr7     3436651-3437214      * |                111</span></span>
<span><span class="co">#&gt;     ...      ...                 ...    ... .                ...</span></span>
<span><span class="co">#&gt;   [239]     chr7 158431413-158433728      * |                111</span></span>
<span><span class="co">#&gt;   [240]     chr7 158818200-158819318      * |                111</span></span>
<span><span class="co">#&gt;   [241]     chr7 158821076-158821876      * |                111</span></span>
<span><span class="co">#&gt;   [242]     chr7 158863108-158864616      * |                111</span></span>
<span><span class="co">#&gt;   [243]     chr7 159015311-159016245      * |                111</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level5">
<h5 id="exporting-overlap-groups">Exporting overlap groups<a class="anchor" aria-label="anchor" href="#exporting-overlap-groups"></a>
</h5>
<p>Each overlap group (e.g., <code>group_100</code>,
<code>group_110</code>, <code>group_111</code>) can be exported for
downstream analysis. The function <code><a href="../reference/exportOverlaps.html">exportOverlaps()</a></code> writes
each group to a Excel file, which makes it easy to reuse the results
outside of R.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># export overlpas</span></span>
<span><span class="fu"><a href="../reference/exportOverlaps.html">exportOverlaps</a></span><span class="op">(</span><span class="va">genomic_overlaps</span>,</span>
<span>               output_dir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>               output_file <span class="op">=</span> <span class="st">"overlap_groups"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] gVenn_0.99.1         GenomicRanges_1.60.0 GenomeInfoDb_1.44.3 </span></span>
<span><span class="co">#&gt; [4] IRanges_2.42.0       S4Vectors_0.46.0     BiocGenerics_0.54.0 </span></span>
<span><span class="co">#&gt; [7] generics_0.1.4      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] eulerr_7.0.4            sass_0.4.10             shape_1.4.6.1          </span></span>
<span><span class="co">#&gt;  [4] polylabelr_0.3.0        stringi_1.8.7           magrittr_2.0.4         </span></span>
<span><span class="co">#&gt;  [7] digest_0.6.37           evaluate_1.0.5          grid_4.5.1             </span></span>
<span><span class="co">#&gt; [10] timechange_0.3.0        RColorBrewer_1.1-3      iterators_1.0.14       </span></span>
<span><span class="co">#&gt; [13] circlize_0.4.16         fastmap_1.2.0           foreach_1.5.2          </span></span>
<span><span class="co">#&gt; [16] doParallel_1.0.17       jsonlite_2.0.0          GlobalOptions_0.1.2    </span></span>
<span><span class="co">#&gt; [19] httr_1.4.7              ComplexHeatmap_2.24.1   UCSC.utils_1.4.0       </span></span>
<span><span class="co">#&gt; [22] codetools_0.2-20        textshaping_1.0.3       jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [25] cli_3.6.5               rlang_1.1.6             crayon_1.5.3           </span></span>
<span><span class="co">#&gt; [28] polyclip_1.10-7         XVector_0.48.0          cachem_1.1.0           </span></span>
<span><span class="co">#&gt; [31] yaml_2.3.10             tools_4.5.1             parallel_4.5.1         </span></span>
<span><span class="co">#&gt; [34] colorspace_2.1-2        GenomeInfoDbData_1.2.14 GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt; [37] vctrs_0.6.5             R6_2.6.1                png_0.1-8              </span></span>
<span><span class="co">#&gt; [40] matrixStats_1.5.0       lifecycle_1.0.4         lubridate_1.9.4        </span></span>
<span><span class="co">#&gt; [43] stringr_1.5.2           fs_1.6.6                clue_0.3-66            </span></span>
<span><span class="co">#&gt; [46] cluster_2.1.8.1         ragg_1.5.0              desc_1.4.3             </span></span>
<span><span class="co">#&gt; [49] pkgdown_2.1.3           bslib_0.9.0             glue_1.8.0             </span></span>
<span><span class="co">#&gt; [52] Rcpp_1.1.0              systemfonts_1.3.1       xfun_0.53              </span></span>
<span><span class="co">#&gt; [55] knitr_1.50              rjson_0.2.23            htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [58] rmarkdown_2.30          compiler_4.5.1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div class="section level4">
<h4 id="example-a549-chip-seq-dataset">Example A549 ChIP-seq dataset<a class="anchor" aria-label="anchor" href="#example-a549-chip-seq-dataset"></a>
</h4>
<ul>
<li>Tav, C., Fournier, É., Fournier, M., Khadangi, F., Baguette, A.,
Côté, M.C., Silveira, M.A.D., Bérubé-Simard, F.-A., Bourque, G., Droit,
A., &amp; Bilodeau, S. (2023). <em>Glucocorticoid stimulation induces
regionalized gene responses within topologically associating
domains.</em> <strong>Frontiers in Genetics</strong>, 14, 1237092.
<a href="https://doi.org/10.3389/fgene.2023.1237092" target="_blank" class="external-link">
doi:10.3389/fgene.2023.1237092 </a>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="supporting-packages">Supporting packages<a class="anchor" aria-label="anchor" href="#supporting-packages"></a>
</h4>
<ul>
<li><p><strong>eulerr</strong> : Larsson, J. (2023). <em>eulerr:
Area-Proportional Euler and Venn Diagrams with Ellipses.</em>
<a href="https://CRAN.R-project.org/package=eulerr" target="_blank" class="external-link">
CRAN package page </a></p></li>
<li><p><strong>ComplexHeatmap</strong> : Gu, Z., Eils, R., &amp;
Schlesner, M. (2016). <em>Complex heatmaps reveal patterns and
correlations in multidimensional genomic data.</em>
<strong>Bioinformatics</strong>, 32(18), 2847–2849.
<a href="https://doi.org/10.1093/bioinformatics/btw313" target="_blank" class="external-link">
doi:10.1093/bioinformatics/btw313 </a></p></li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://ckntav.github.io/" class="external-link">Christophe Tav</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
